ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -11,35 +11,167 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { PanelProps, useTimeSeriesQueries } from '@perses-dev/plugin-system';
+import { useMemo, useState } from 'react';
+import { merge } from 'lodash-es';
+import { PanelProps, useTimeSeriesQueries, useTimeRange } from '@perses-dev/plugin-system';
+import { GridComponentOption } from 'echarts';
+import { Box, Skeleton } from '@mui/material';
+import { LineChart, EChartsDataFormat, ZoomEventData, Legend } from '@perses-dev/components';
 import { useSuggestedStepMs } from '../../model/time';
-import { TimeSeriesChartOptions } from './time-series-chart-model';
-import { TimeSeriesChartContainer } from './TimeSeriesChartContainer';
+import { StepOptions, ThresholdColors, ThresholdColorsPalette } from '../../model/thresholds';
+import { TimeSeriesChartOptions, DEFAULT_LEGEND } from './time-series-chart-model';
+import { getLineSeries, getCommonTimeScale, getYValues, getXValues, EMPTY_GRAPH_DATA } from './utils/data-transform';
+import { getRandomColor } from './utils/palette-gen';
 
 export type TimeSeriesChartProps = PanelProps<TimeSeriesChartOptions>;
 
 export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
   const {
-    spec: { queries, show_legend, thresholds, unit },
+    spec: { queries, thresholds },
     contentDimensions,
   } = props;
 
+  // popuate 'position' and 'show' if undefined
+  const legend = merge({}, DEFAULT_LEGEND, props.spec.legend);
+
+  const unit = props.spec.unit ?? {
+    kind: 'Decimal',
+    decimal_places: 2,
+  };
```

Comment:
Could do DEFAULT_UNIT too (I like how you did DEFAULT_LEGEND a lot), but really whatevs / potato potato

Reply:
Glad to hear you like the merge approach for DEFAULT_LEGEND, I'll update unit to be similar in a follow up PR when I go to make that property editable next!